api/src/test/java/org/openmrs/AllergiesTest.java

```diff
@@ -330,15 +332,13 @@ public void addAll_shouldNotAllowDuplicateNonCodedAllergen(){
 		
 		Allergies allergies = new Allergies();
 		allergies.add(allergy1);
-		allergies.add(allergy2);
-		
-		allergies.addAll(allergies);
+		assertThrows(APIException.class, () -> allergies.add(allergy2));
```

Comment:
The last statement in the tests testing exceptions with `@Test(expected = xxx)` should be the ones throwing the exception. It can happen that they are not throwing an exception and that this was hidden from us by a statement before that was throwing the exception. If that is the case `assertThrows` should now highlight such cases.